crates/edr_provider/src/error.rs

```diff
@@ -76,14 +82,4 @@ pub enum ProviderError {
     /// The address is not owned by this node.
     #[error("{address} is not owned by this node")]
     UnknownAddress { address: Address },
-    /// Block number doesn't exist in blockchain
```

Comment:
I replaced these with a single `InvalidBlockNumberOrHash` as we only check if any of these match to return `None` in the RPC and there will be many methods making this check.